09-features.md

```diff
@@ -26,20 +26,20 @@ The Context column decodes as follows:
-
+| Bits  | Name                             | Description                                               | Context  | Dependencies              | Link                                                    |
+|-------|----------------------------------|-----------------------------------------------------------|----------|---------------------------|---------------------------------------------------------|
+| 0/1   | `option_data_loss_protect`       | Requires or supports extra `channel_reestablish` fields   | IN       |                           | [BOLT #2][bolt02-retransmit]                            |
+| 3     | `initial_routing_sync`           | Sending node needs a complete routing information dump    | I        |                           | [BOLT #7][bolt07-sync]                                  |
+| 4/5   | `option_upfront_shutdown_script` | Commits to a shutdown scriptpubkey when opening channel   | IN       |                           | [BOLT #2][bolt02-open]                                  |
+| 6/7   | `gossip_queries`                 | More sophisticated gossip control                         | IN       |                           | [BOLT #7][bolt07-query]                                 |
+| 8/9   | `var_onion_optin`                | Requires/supports variable-length routing onion payloads  | IN9      |                           | [Routing Onion Specification][bolt04]                   |
+| 10/11 | `gossip_queries_ex`              | Gossip queries can include additional information         | IN       | `gossip_queries`          | [BOLT #7][bolt07-query]                                 |
+| 12/13 | `option_static_remotekey`        | Static key for remote output                              | IN       |                           | [BOLT #3](03-transactions.md)                           |
+| 14/15 | `payment_secret`                 | Node supports `payment_secret` field                      | IN9      | `var_onion_optin`         | [Routing Onion Specification][bolt04]                   |
+| 16/17 | `basic_mpp`                      | Node can receive basic multi-part payments                | IN9      | `payment_secret`          | [BOLT #4][bolt04-mpp]                                   |
+| 18/19 | `option_support_large_channel`   | Can create large channels                                 | IN       |                           | [BOLT #2](02-peer-protocol.md#the-open_channel-message) |
+| 20/21 | `option_anchor_outputs`          | Anchor outputs                                            | IN       | `option_static_remotekey` | [BOLT #3](03-transactions.md)                           |
+| 24/25 | `zlib_compress`                  | Support for zlib compression                              | IN       | `gossip_queries_ex`       | [BOLT #7][bolt07-query]                                 |
```

Comment:
How do we interpret mandatory/optional bits ? If you setup the optional bit, does it signal as "I accept compressed queries" and implicitly signal raw format acceptance but doesn't enforce a preference on scids query sender ?

But if you're an upgraded node which don't support zlib but still want to signal raw format acceptance how do you do to dissociate yourself from non-upgraded nodes ?

Reply:
Good point.

I'd say that optional support means I can do zlib and will comply with what senders ask for (and may ask you for zlib if you support it as well).

Mandatory support means I'll only do zlib and never uncompressed data. Is that too hardcore? I think it makes sense to have this hardcore mode, and it's up to node operators to decide whether they want to go hardcore or not (implementations will very likely only default to optional support or no support).

Reply:
I really agree it's up to node operators to decide what their gossip queries format, let's be conservative with your bandwidth.

But still, after this PR, how a queried node should signal "I-don't-want-to-support-zlib-but-only-raw-format" to dissociate themselves from non-upgraded nodes from a querying node viewpoint ?

Maybe we discuss reversing the feature-bit to `option_raw_format` at next meeting, it sounds to solve the issue IMO.

Reply:
I'd suggest using a `format_raw` bit instead - if a user is concerned about bandwidth, there should be a global bandwidth limiter provided which limits bandwidth of pings, syncs, and everything else. Not just enforcing that syncs only happen via one mechanism, which only indirectly limits bandwidth and will fall over if you get an old peer sending lots of pings or a sudden influx of peers.

Reply:
I understand the reasoning behind reversing this to an `option_raw_format` / `format_raw`, but I find it weird to have feature bits that actually announce absence of support for something. Feature bits should announce support for additional capabilities, not reduced capabilities.

Granted, we're in this situation because we chose at the time to make zlib support mandatory, but I'm not sure we should go in that direction. I'd like to hear @rustyrussell's opinion on this, he and @sstone did the initial zlib/gossip_queries work.

Reply:
I went for a slightly different approach in https://github.com/lightningnetwork/lightning-rfc/pull/825/commits/673c912e1a01c317c9d82f8e8b2cbe7a1f3c66c9